index.js

```diff
@@ -113,4 +117,76 @@ app.get('/squadStats', async (req, res) => {
 
     res.send(retVal);
 
-});
\ No newline at end of file
+});
+
+//stats squad individual
+app.get('/squadStatsIndiv', async (req, res) => {
+
+    let gt = req.query.gt;
+
+    let result = await lib.halo.mcc['@0.0.11'].squad.activity({
+        gamertags: [
+            'sagwasswagga',
+            'Cragle 3',
+            'A Pregnant Nunn',
+            'babaganoosh069',
+            'Girthy Jello',
+            'SirPuffingtin',
+            'ya boi nar',
+            'l swordzzz'
+        ],
+        minSquadSize: '3',
+        game: 'Halo 3'
+    });
+
+
+    let games = result.games;
+
+    let aggregSqdStats = [];
+    
+    for(let match of games) {
+        let win = match.squadWon;
+        let avgHeadshotRate = 0.0;
+        let avgKD = 0.0;
+        let avgScore = 0.0;
+        let avgKills = 0.0;
+        let avgDeaths = 0.0;
+        let avgAssists = 0.0;
+        let playedAt = match.playedAt;
+
+        for(let stat of match.stats) {
+            if(stat.gamertag.toUpperCase() == gt.toUpperCase()) {
```

Comment:
`gt` is for a single gt in `req.query` explain this more please bc to me it seems this will only be for one person. but you make it seem this look will take average for entire squad

Reply:
okay yeah this makes sense, so the idea behind this request is to determine how well we do on CT specifically. The individual query "/myStats" can just get any number of recent games (limit 100), but say on that thursday I have a heavy session where i play like 10 games and get a 10.0 KD (for sake of example im not that sick), so it would be annoying to have to loop over the most recent n games and check all stats for matching gamertags from the squad and exclude any games that does not have the squad in it to remove that 10.0KD from my CT specific performance.

Therefore, I thought it would be easier to query from the squads endpoint and filter out everyone else that was not queried for, so we wouldnt need to do any averaging either.

Reply:
no no not all of them are required, there is a parameter for the squad query "minSquadSize" that will return a game if it matches at minimum minSquadSize number of gamertags that we request, and it will tell us who is present & absent, checkout this response when I was logging it before:

![image](https://user-images.githubusercontent.com/17136891/117586239-be26de00-b0cb-11eb-9e8f-940110d82a4c.png)

Reply:
in this case, minSquadSize is always 3, but I can make it a query param, that would be a good idea

Reply:
ok resp